src/main/java/com/yscope/clp/compressorfrontend/MessageDecoder.java

```diff
@@ -44,40 +55,54 @@ private native void setVariableHandlingRuleVersions (
    * @throws IOException if decoding fails
    */
   public String decodeMessage(
-      String logtype,
+      @NotNull String logtype,
       String[] dictionaryVars,
       long[] encodedVars
   ) throws IOException {
-    // Flatten dictionaryVars (["var1", "var2", ...] -> "var1var2...")
-    int[] dictionaryVarEndOffsets = new int[dictionaryVars.length];
-    ByteArrayOutputStream dictionaryVarsOutputStream = new ByteArrayOutputStream();
-    int lastDictionaryVarEndOffset = 0;
-    for (int i = 0; i < dictionaryVars.length; ++i) {
-      byte[] dictionaryVarBytes = dictionaryVars[i].getBytes(StandardCharsets.UTF_8);
-      dictionaryVarsOutputStream.write(dictionaryVarBytes);
-      lastDictionaryVarEndOffset += dictionaryVarBytes.length;
-      dictionaryVarEndOffsets[i] = lastDictionaryVarEndOffset;
+    Objects.requireNonNull(logtype);
+
+    int[] dictionaryVarEndOffsets = null;
+    byte[] allDictionaryVars = null;
+    if (null != dictionaryVars) {
+      // Flatten dictionaryVars (["var1", "var2", ...] -> "var1var2...")
+      dictionaryVarEndOffsets = new int[dictionaryVars.length];
+      ByteArrayOutputStream dictionaryVarsOutputStream = new ByteArrayOutputStream();
+      int lastDictionaryVarEndOffset = 0;
+      for (int i = 0; i < dictionaryVars.length; ++i) {
+        byte[] dictionaryVarBytes = dictionaryVars[i].getBytes(StandardCharsets.UTF_8);
+        dictionaryVarsOutputStream.write(dictionaryVarBytes);
+        lastDictionaryVarEndOffset += dictionaryVarBytes.length;
+        dictionaryVarEndOffsets[i] = lastDictionaryVarEndOffset;
+      }
+      allDictionaryVars = dictionaryVarsOutputStream.toByteArray();
     }
 
-    byte[] messageBytes = decodeMessageNative(logtype.getBytes(StandardCharsets.ISO_8859_1),
-        dictionaryVarsOutputStream.toByteArray(), dictionaryVarEndOffsets, encodedVars);
+    byte[] logtypeBytes = logtype.getBytes(StandardCharsets.ISO_8859_1);
+    byte[] messageBytes = decodeMessageNative(
+        logtypeBytes, logtypeBytes.length, allDictionaryVars,
+        null == allDictionaryVars ? 0 : allDictionaryVars.length, dictionaryVarEndOffsets,
```

Comment:
nit: I feel it's little hard to read the line since it contains a ternary operator plus an extra variable. can we make ternary operator to be on its own line?